kernels/volk/volk_32fc_s32f_power_spectrum_32f.h

```diff
@@ -190,23 +193,25 @@ volk_32fc_s32f_power_spectrum_32f_generic(float* logPowerOutput, const lv_32fc_t
                                           const float normalizationFactor, unsigned int num_points)
 {
   // Calculate the Power of the complex point
-  const float* inputPtr = (float*)complexFFTInput;
-  float* realFFTDataPointsPtr = logPowerOutput;
-  const float iNormalizationFactor = 1.0 / normalizationFactor;
-  unsigned int point;
-  for(point = 0; point < num_points; point++){
-    // Calculate dBm
-    // 50 ohm load assumption
-    // 10 * log10 (v^2 / (2 * 50.0 * .001)) = 10 * log10( v^2 * 10)
-    // 75 ohm load assumption
-    // 10 * log10 (v^2 / (2 * 75.0 * .001)) = 10 * log10( v^2 * 15)
+  const float normFactSq = 1.0 / (normalizationFactor * normalizationFactor);
 
-    const float real = *inputPtr++ * iNormalizationFactor;
-    const float imag = *inputPtr++ * iNormalizationFactor;
+  // Calculate dBm
+  // 50 ohm load assumption
+  // 10 * log10 (v^2 / (2 * 50.0 * .001)) = 10 * log10( v^2 * 10)
+  // 75 ohm load assumption
+  // 10 * log10 (v^2 / (2 * 75.0 * .001)) = 10 * log10( v^2 * 15)
 
-    *realFFTDataPointsPtr = 10.0*log10f(((real * real) + (imag * imag)) + 1e-20);
-    realFFTDataPointsPtr++;
-  }
+  // Calc mag^2
+  volk_32fc_magnitude_squared_32f(logPowerOutput, complexFFTInput, num_points);
+
+  // Finish ((real * real) + (imag * imag)) calculation:
+  volk_32f_s32f_multiply_32f(logPowerOutput, logPowerOutput, normFactSq, num_points);
+
+  // The following calculates 10*log10(x) = 10*log2(x)/log2(10) = (10/log2(10))
+  // * log2(x)
+  volk_32f_log2_32f(logPowerOutput, logPowerOutput, num_points);
+  volk_32f_s32f_multiply_32f(logPowerOutput, logPowerOutput, volk_log2to10factor,
+                             num_points);
```

Comment:
I like the "meta" approach here.

Since this is the generic kernel, I wonder if we should call the generic kernels here as well.

Reply:
We'd lose all the performance benefits if we forcibly call the generics.  I think the benefit here is that this generic calls the best-performing version of each of the called kernels.

Reply:
That's true. Also, a combination of different kernels for different parts might be beneficial in terms of performance. In this case, I wonder if we should just remove all the other kernels. Or go for a generic simple implemenentation and a generic optimized version that just calls into the other kernels.